src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -4973,6 +4981,22 @@ public Expression GenerateCallContains(Expression lhs, Expression rhs, bool igno
                 rhs.Cast(typeof(object)));
         }
 
+        public object VisitTernaryExpression(TernaryExpressionAst ternaryExpressionAst)
+        {
+            var tmp = NewTemp(typeof(object), "result");
+
+            var expr = Expression.IfThenElse(
```

Comment:
Using `Expression.Condition` here would let you skip the temp/block.

Reply:
True, especially since both `IfTrue` and `IfFalse` sides are cast to object type.
Will make the change.